contracts/Lock.sol

```diff
@@ -117,25 +123,52 @@ contract Lock is AragonApp, IForwarder, IForwarderFee {
     /**
     * @notice Locks the required amount of tokens and executes the specified action
     * @dev IForwarder interface conformance. Consider using pretransaction on UI for necessary approval.
+    *      We check for edge cases where an approve is made and before this function is called, one of msg.sender current locks is unlocked resulting in a different griefing amount.
     * @param _evmCallScript Script to execute
     */
-    function forward(bytes _evmCallScript) public {
+    function forward(bytes _evmCallScript) public  auth(LOCK_TOKENS_ROLE) {
         require(canForward(msg.sender, _evmCallScript), ERROR_CAN_NOT_FORWARD);
 
+        (uint256 griefAmount, uint256 griefDuration) = getGriefing(msg.sender);
+
+        uint256 allowance = token.allowance(msg.sender, address(this));
+        uint256 computedAmount = lockAmount.add(griefAmount);
+
+        uint256 totalAmount = allowance > computedAmount ? allowance : computedAmount;
+        uint256 totalDuration = lockDuration.add(griefDuration);
+
         WithdrawLockLib.WithdrawLock[] storage addressWithdrawLocks = addressesWithdrawLocks[msg.sender];
-        uint256 unlockTime = getTimestamp().add(lockDuration);
-        addressWithdrawLocks.push(WithdrawLockLib.WithdrawLock(unlockTime, lockAmount));
+        uint256 unlockTime = getTimestamp().add(totalDuration);
+        addressWithdrawLocks.push(WithdrawLockLib.WithdrawLock(unlockTime, totalAmount));
 
-        require(token.safeTransferFrom(msg.sender, address(this), lockAmount), ERROR_TRANSFER_REVERTED);
+        require(token.safeTransferFrom(msg.sender, address(this), totalAmount), ERROR_TRANSFER_REVERTED);
 
-        emit NewLock(msg.sender, unlockTime, lockAmount);
+        emit NewLock(msg.sender, unlockTime, totalAmount);
         runScript(_evmCallScript, new bytes(0), new address[](0));
     }
 
     function getWithdrawLocksCount(address _lockAddress) public view returns (uint256) {
         return addressesWithdrawLocks[_lockAddress].length;
     }
 
+    /**
+    * @notice Get's amount and duration penalty based on the number of current locks `_sender` has
+    * @param _sender account that is going to lock tokens
+    * @return amount and duration penalty
+    */
+    function getGriefing(address _sender) public view returns (uint256, uint256) {
+        WithdrawLockLib.WithdrawLock[] memory addressWithdrawLocks = addressesWithdrawLocks[_sender];
+
+        uint256 activeLocks = 0;
+        for (uint256 withdrawLockIndex = 0; withdrawLockIndex < addressWithdrawLocks.length; withdrawLockIndex++) {
+            if (getTimestamp() < addressWithdrawLocks[withdrawLockIndex].unlockTime) {
+                activeLocks += 1;
+            }
+        }
+
+        return (lockAmount.mul(activeLocks), lockDuration.mul(activeLocks));
```

Comment:
Just a thought but we could include the ability to set a `griefing factor` value which can be set via a new permission. The griefing factor could just be a whole number eg 0, 1, 2 etc meaning when set to 0, there is no griefing, 1 is as above, 2 is worse etc. The above would then look like this: 
```
return (lockAmount.mul(activeLocks).mul(griefingFactor), lockDuration.mul(activeLocks).mul(griefingFactor));
```
Alternatively the griefing factor could be fractional by specifying a `wholeGriefingValue` of say 100 (or other 10^x value) as a constant in the contract and setting the `griefingFactor` effectively as a % of this. Eg a `griefingFactor` of 50 would be the equivalent of 0.5 in the above example, 100 would be 1, same as above, 200 would be 2. The above would then look like this: 
```
return (lockAmount.mul(activeLocks).mul(griefingFactor).div(wholeGriefingValue), lockDuration.mul(activeLocks).mul(griefingFactor).div(wholeGriefingValue));
```

Reply:
I like the second approach better. Think makes more sense to have fractional factor.

And we could show the griefing factor as a % on the UI